src/v/kafka/server/handlers/fetch.cc

```diff
@@ -484,29 +485,23 @@ static std::vector<shard_fetch> group_requests_by_shard(op_context& octx) {
  */
 
 static ss::future<> fetch_topic_partitions(op_context& octx) {
-    std::vector<ss::future<>> fetches;
-    fetches.reserve(ss::smp::count);
-
-    ss::shard_id shard = 0;
-    for (auto& shard_fetch : group_requests_by_shard(octx)) {
-        fetches.push_back(
-          handle_shard_fetch(shard++, octx, std::move(shard_fetch)));
-    }
-
-    return ss::do_with(
-      std::move(fetches), [&octx](std::vector<ss::future<>>& fetches) {
-          return ss::when_all_succeed(fetches.begin(), fetches.end())
-            .then([&octx] {
-                if (octx.should_stop_fetch()) {
-                    return ss::now();
-                }
-                octx.reset_context();
-                // debounce next read retry
-                return ss::sleep(std::min(
-                  config::shard_local_cfg().fetch_reads_debounce_timeout(),
-                  octx.request.data.max_wait_ms));
-            });
-      });
+    auto planner = make_fetch_planner<simple_fetch_planner>();
+
+    auto fetch_plan = planner.create_plan(octx);
+
+    fetch_plan_executor executor
+      = make_fetch_plan_executor<parallel_fetch_plan_executor>();
+    co_await executor.execute_plan(octx, std::move(fetch_plan));
```

Comment:
where does the core local data get updated?

Reply:
It is when we receive read result. In here: https://github.com/vectorizedio/redpanda/blob/db2cf5626b74318bccd7e673ff99bb2d9b60b2a4/src/v/kafka/server/handlers/fetch.cc#L282